### ISSUE TITLE
I would like to change the syntax highlighting colours for file previews.

### ISSUE THREAD

**Issue by krompus** [ranger/ranger#547]
This isn't necessarily a problem with ranger itself, but I'd still like to figure this out.

I had a look at ~/.config/ranger/scope.sh, and found the relevant lines:

```
# Syntax highlight for text files:
text/* | */xml)
    try safepipe highlight --out-format=ansi "$path" && { dump | trim; exit 5; }
    try safepipe pygmentize "$path" && { dump | trim; exit 5; }
    exit 2;;
```

But when I try using `highlight`'s `--style=matrix` (for example), it still shows the default colourscheme.

I edited the highlight line to this:
    `try safepipe highlight --style=matrix --out-format=ansi "$path" && { dump | trim; exit 5; }`

I've tried with a few different style names, and I've fiddled with that line. No dice.

You can list the available `highlight` themes with `ls /usr/share/highlight/themes/`, by the way.

Any ideas how to get it working?

**Comment by Vifon** [comment]
I vaguely recall something about the themes not working with the ANSI output format. They work properly when exporting to HTML. There is not much you can do about it.

**Comment by krompus** [comment]
Ah, alright then. Thanks anyways!

**Comment by Vifon** [comment]
You may want to try with pygments. I've just found an IRC conversation from November that confirms that highlight cannot use themes in this case and it was considered if pygments can do so. Nothing specific though so be sure to tell us about it if you will get it to work. ;)

**Comment by krompus** [comment]
Sweet.

Yeah, I fiddled with pygments a bit, but I couldn't get it working.

I'll update if I figure it out!

**Comment by mrogalski** [comment]
I have managed to set it up by:

-  putting the following into `~/.config/ranger/scope.sh`:
```bash 
text/* | */xml)
        try highlight --out-format=ansi "$path" && { dump | trim; exit 5; }
        try pygmentize -f 256 -O style=monokai "$path" && { dump | trim; exit 5; }
        exit 2;;
``` 

- setting the TERM variable to xterm-256color.  Try it, like so:
```bash
   export TERM='xterm-256color'
   clear 
```
This might fail with  `'xterm-256color': unknown terminal type.`.  If so, check with Your ncurses installation -- the terminfo for xterm-256color should be in `/usr/share/terminfo/x/xterm-256color `or Your distros equivalent.

If it works You can put `export TERM='xterm-256color'` into Your shells rc file.

**Comment by krompus** [comment]
ERMAHGERD OSSERM!

You can get a list of all available styles with `pygmentize -L`. Quite frankly, many of them are angry fruit salad; it's a shame there's no jellybeans. Maybe I'll make it someday.

Thanks bru!

**Comment by fd2** [comment]
For highlight, this works well on my box:
In `scope.sh` use:   `try highlight --out-format=xterm256 ...` 
and in your environment:   `export TERM='xterm-256color`

**Comment by krompus** [comment]
@fd2 yep, that works, but it looks like [hot vomit.](https://ptpb.pw/hK8b.png) (highlight on the left, pygmentize on the right)

I tried changing the theme by adding `--config-file=fruit`, but that didn't work.

**Comment by fd2** [comment]
@krompus: Using `--style=<style name>` rather than `--config-file=<style name>` might give better results.
e.g. `try highlight --out-format=xterm256 --style=zenburn "$path"`

**Comment by krompus** [comment]
Ah! That works!

I'll fiddle with this; thank you!

**Comment by piggyslasher** [comment]
Actually, I just want to add a fix that worked for me.

Change

```shell
highlight_format=xterm256
```

to

```shell
highlight_format=ansi
```

in your `scope.sh` and reload.

**Comment by ha7ilm** [comment]
TL;DR what I had to do at `ranger 1.9.3` is just the following: 

1. get a list of themes:

    ls /usr/share/highlight/themes/

2. add `--style=something` (without the `.theme` extension) into `scope.sh`:

![image](https://user-images.githubusercontent.com/5903408/116680709-47dbfa80-a9ac-11eb-8859-56ad1708e536.png)

**Comment by toonn** [comment]
@ha7ilm, you could also simply set `HIGHLIGHT_STYLE=xoria256` in your shell's environment. That way you don't need to copy and alter `scope.sh` and you'll automatically benefit from changes we make to it.

Though it looks like you may be on Manjaro? They include a *very* anachronistic version of `scope.sh` for some reason. So you might want to just get the one from the ranger repo.

**Comment by ha7ilm** [comment]
Thanks for the ideas! (I'm using Ubuntu though.)

**Comment by toonn** [comment]
Oh no, ubuntu includes an old `scope.sh` as well? Why 😩

**Comment by Zeioth** [comment]
Thank you, it looks great
![monokai](https://user-images.githubusercontent.com/3357792/117861252-85386600-b291-11eb-8070-6685c259f756.png)

(Yes, I've disabled scary colors on my term)

**Comment by fynnfluegge** [comment]
hey @Zeioth how did you get those border separators around the views?

**Comment by Zeioth** [comment]
@fynnfluegge im not 100% what you mean, but if you refer yo having only 2 columnd you can use this on your rc.conf

    set column_ratios 2,4

**Comment by fynnfluegge** [comment]
Thanks for responding fast @Zeioth. No I don't mean the tiling ratios, I mean the borders which create this box-like view for each column 😅

**Comment by rogalski-wmii-uni-lodz-pl** [comment]
Set draw_borders to "outline", "separators" or "both", see which You like the most.

**Comment by fynnfluegge** [comment]
Ah thanks got it, not sure how I didn't saw this in the conf file 🤦‍♂️

**Comment by toonn** [comment]
I'd like to ask people to keep issue comments to strictly related things. Please use the r/ranger subreddit for questions like how to achieve a certain look. It's a lot easier (and more appropriate) for the community to engage there.

**Comment by wfpaisa** [comment]
ranger uses bat to display the files, you can see a list of bat themes: 

```sh
$ bat --list-themes
```
 and the `scope.sh`
add the --theme 
![image](https://github.com/ranger/ranger/assets/1735752/c5ea36e2-02a4-4957-9db7-8130b988cf11)
